src/transformers/tokenization_utils_base.py

```diff
@@ -2199,6 +2199,11 @@ def _from_pretrained(
 
                 if isinstance(token, AddedToken):
                     added_tokens_decoder[int(idx)] = token
+                    if str(token) in additional_special_tokens:
+                        # at this point if the token is in `additional_special_tokens` as an str, should be updated
+                        additional_special_tokens.remove(str(token))
```

Comment:
Only use the default legacy values for AddedToken if the token is not already in the added tokens decoder